### PR TITLE
Add AnnotationType for indicator buttons and RCA

### PR DESCRIPTION
**Pull request by stephanheilner**

### CODE REVIEW

**Comment by stephanheilner** [comment]
Tests pass locally

File: LDSAnnotations/AnnotationStore+Annotation.swift [hunk 1]
```diff
@@ -316,6 +316,63 @@ public extension AnnotationStore {
...
+            }).flatMap { (annotationID, annotations) -> (annotation: Annotation, type: AnnotationType)? in
+                guard let annotationWithType = annotations.first else { return nil }
+                
+                if annotations.count > 1 {
+                    return (annotation: annotationWithType.annotation, type: .multiple)
+                }
+                return (annotation: annotationWithType.annotation, type: annotationWithType.type)
+            }
+            
+            let linkDestinationAnnotationsWithType = try db.prepare(AnnotationTable.table.select(distinct: AnnotationTable.table[*]).join(LinkTable.table, on: AnnotationTable.table[AnnotationTable.id] == LinkTable.annotationID).filter(LinkTable.table[LinkTable.docID] == docID)).map { row -> (annotation: Annotation, type: AnnotationType) in
```
Comment (nickmshelley) [inline LDSAnnotations/AnnotationStore+Annotation.swift]: Does this work with multiple links in a single annotation?

**Comment by nickmshelley** [comment]
👍

**Comment by stephanheilner** [comment]
Tests still pass locally

**Pull request merged**